modules/db/src/main/scala/gem/dao/StepDao.scala

```diff
@@ -535,6 +305,207 @@ object StepDao {
         VALUES ($oid, $loc, ${i.instrument: Instrument}, ${t} :: step_type)
       """.update
 
-  }
+    object F2 {
+
+      def selectAll(oid: Observation.Id): Query0[(Loc, DynamicConfig.F2)] =
+        sql"""
+          SELECT s.location,
+                 i.disperser,
+                 i.exposure_time,
+                 i.filter,
+                 i.fpu,
+                 i.lyot_wheel,
+                 i.read_mode,
+                 i.window_cover
+            FROM step s
+                 LEFT OUTER JOIN step_f2 i
+                   ON i.step_f2_id = s.step_id
+           WHERE s.observation_id = $oid
+        """.query[(Loc, DynamicConfig.F2)]
+
+      def selectOne(oid: Observation.Id, loc: Loc): Query0[DynamicConfig.F2] =
+        sql"""
+          SELECT i.disperser,
+                 i.exposure_time,
+                 i.filter,
+                 i.fpu,
+                 i.lyot_wheel,
+                 i.read_mode,
+                 i.window_cover
+            FROM step s
+                 LEFT OUTER JOIN step_f2 i
+                   ON i.step_f2_id = s.step_id
+           WHERE s.observation_id = $oid AND s.location = $loc
+        """.query[DynamicConfig.F2]
+
+      def insert(id: Int, f2: DynamicConfig.F2): Update0 =
+        sql"""
+          INSERT INTO step_f2 (
+            step_f2_id,
+            disperser, exposure_time, filter, fpu, lyot_wheel, read_mode, window_cover
+          )
+          VALUES (
+            $id,
+            ${f2.disperser},
+            ${f2.exposureTime},
+            ${f2.filter},
+            ${f2.fpu},
+            ${f2.lyotWheel},
+            ${f2.readMode},
+            ${f2.windowCover})
+        """.update
+    }
+
+    object Gmos {
 
+      import gem.config.Gmos.{GmosCommonDynamicConfig, GmosCustomMask, GmosGrating}
+      import DynamicConfig.{GmosNorth, GmosSouth}
+
+      final case class GmosGratingBuilder[D](
+                         disperser: Option[D],
+                         disperserOrder: Option[GmosDisperserOrder],
+                         wavelength: Option[Int]) {
+
+
+        def toGrating: Option[GmosGrating[D]] =
+          for {
+            d <- disperser
+            o <- disperserOrder
+            w <- wavelength.flatMap(Wavelength.fromAngstroms)
+          } yield GmosGrating(d, o, w)
+      }
+
+      final case class GmosFpuBuilder[U](
+                         mdfFileName: Option[String],
+                         customSlitWidth: Option[GmosCustomSlitWidth],
+                         builtin: Option[U]) {
+
+        val toFpu: Option[GmosCustomMask \/ U] = {
```

Comment:
`EitherT` uses `\/` so we're arguably better off with `\/` until we switch to cats. It will all get resolved soonish, I hope.